app/controllers/masquerade_controller.rb

```diff
@@ -8,7 +8,11 @@ class MasqueradeController < UserController
   private
 
     def verify_deputies
-      return if primary_user.available_deputy?(current_user)
+      if current_user.masquerading?
+        return if primary_user.available_deputy?(current_user.impersonator)
+      elsif primary_user.available_deputy?(current_user)
+        return
+      end
```

Comment:
I might do something like:

``` ruby
def current_deputy
  if current_user.masquerading?
    current_user.impersonator
  else
    current_user
end
```

Then the change is just: `return if primary_user.available_deputy?(current_deputy)`

I'm also wondering if this logic should be in `UserDecorator` like:

``` ruby
def as_deputy
  return self unless masquerading?

  self.impersonator
end
```